python-checks/src/main/java/org/sonar/python/checks/DeadStoreCheck.java

```diff
@@ -65,15 +72,28 @@ public void initialize(Context context) {
    * Bottom-up approach, keeping track of which variables will be read by successor elements.
    */
   private static void verifyBlock(SubscriptionContext ctx, CfgBlock block, LiveVariablesAnalysis.LiveVariables blockLiveVariables,
-                                  Set<Symbol> readSymbols, FunctionDef functionDef) {
+    Set<Symbol> readSymbols, FunctionDef functionDef) {
 
     DeadStoreUtils.findUnnecessaryAssignments(block, blockLiveVariables, functionDef)
       .stream()
       // symbols should have at least one read usage (otherwise will be reported by S1481)
       .filter(unnecessaryAssignment -> readSymbols.contains(unnecessaryAssignment.symbol))
       .filter((unnecessaryAssignment -> !isException(unnecessaryAssignment.symbol, unnecessaryAssignment.element, functionDef)))
-      .forEach(unnecessaryAssignment ->
-        ctx.addIssue(unnecessaryAssignment.element, String.format(MESSAGE_TEMPLATE, unnecessaryAssignment.symbol.name())));
+      .forEach(unnecessaryAssignment -> {
+        IssueWithQuickFix issue;
+        Optional<Token> separatorToken = Optional.of(unnecessaryAssignment.element)
+          .filter(AssignmentStatementImpl.class::isInstance)
+          .map(x -> ((AssignmentStatementImpl) x).separator())
+          .filter(x -> !Objects.equals(x.value(), "\n"));
+        if (separatorToken.isPresent()) {
+          issue = (IssueWithQuickFix) ctx.addIssue(unnecessaryAssignment.element.firstToken(), separatorToken.get(),
+            String.format(MESSAGE_TEMPLATE, unnecessaryAssignment.symbol.name()));
+        } else {
+          issue = (IssueWithQuickFix) ctx.addIssue(unnecessaryAssignment.element,
+            String.format(MESSAGE_TEMPLATE, unnecessaryAssignment.symbol.name()));
+        }
```

Comment:
I've actually tried to do something similar (less clean) but it raised 2 issues instead of 1 on the line 323: `def dead_store(): unused = 24; unused = 42; print(unused);` 
My guess is that the `orElse()` function somehow creates the possible issue which is automatically added to the context.
I found that using `orElseGet(() -> ctx.addIssue(element, message))` doesn't raise twice an issue. Thanks for the suggestion and the clean code !